src/main/java/org/sonarlint/intellij/core/ServerIssueUpdater.java

```diff
@@ -140,4 +112,51 @@ public void disposeComponent() {
       LOGGER.debug("rejected " + rejected.size() + " pending tasks");
     }
   }
+
+  private class IssueUpdateRunnable implements Runnable {
+    private final SonarQubeServer server;
+    private final VirtualFile virtualFile;
+    private final ConnectedSonarLintEngine engine;
+    private final String moduleKey;
+    private final String relativePath;
+
+    private IssueUpdateRunnable(SonarQubeServer server, VirtualFile virtualFile, ConnectedSonarLintEngine engine, String moduleKey, String relativePath) {
+      this.server = server;
+      this.virtualFile = virtualFile;
+      this.engine = engine;
+      this.moduleKey = moduleKey;
+      this.relativePath = relativePath;
+    }
+
+    @Override public void run() {
+      try {
+        Iterator<ServerIssue> serverIssues = fetchServerIssues(server, engine, moduleKey, relativePath);
+        Collection<Trackable> serverIssuesTrackable = toStream(serverIssues).map(ServerIssueTrackable::new).collect(Collectors.toList());
+
+        if (!serverIssuesTrackable.isEmpty()) {
+          issueManager.matchWithServerIssues(virtualFile, serverIssuesTrackable);
+        }
+      } catch (Throwable t) {
+        // note: without catching Throwable, any exceptions raised in the thread will not be visible
+        console.error("error while fetching and matching server issues", t);
+        t.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1148)